src/containerapp/azext_containerapp/_up_utils.py

```diff
@@ -1214,8 +1214,8 @@ def _get_env_and_group_from_log_analytics(
                 resource_group.name = env_details["resource_group"]
 
 
-def _get_acr_from_image(cmd, app):
-    if app.image is not None and "azurecr.io" in app.image:
+def _get_acr_from_image(cmd, app, registry_server_set=None):
+    if app.image is not None and "azurecr.io" in app.image and not registry_server_set:
```

Comment:
Can we skip calling this `_get_acr_from_image` method if `registry_server_set` is false in the `_set_up_defaults` method instead?

Reply:
Not sure I fully understand. We get ACR from the image only if `registry_server_set` is false, correct?

Reply:
actually second thought, i think refactoring it would be better. sorry for the confusion. updating it in the next commit

Reply:
Thank you @harryli0108!